### PR TITLE
Feature/OB-6

### PR DESCRIPTION
**Pull request by ppseprus**

### CODE REVIEW

File: app/actions/app-data.js [hunk 1]
```diff
@@ -21,11 +21,14 @@ export function updateUser({user, id}) {
...
-export function fetchData() {
+export function fetchData(username) {
+    if (username === undefined) {
+        // TODO: implement error handling
+    }
     return function (dispatch) {
         dispatch(fetchingData());
 
-        return getUsers()
+        return getUsers(username)
```
Comment (csabbee) [inline app/actions/app-data.js]: I would rather create a separate method, getUser. As this is should be for getting all the users.
Well we can argue though whether this method should exist...

File: app/reducers/app-data.js [hunk 2]
```diff
@@ -5,28 +5,28 @@ const defaultState = {
...
-                    if (user.id === action.user.id) {
-                        return action.user;
-                    } else {
-                        return user;
-                    }
-                })
-            };
-        default:
-            return state;
+    case 'FETCHING_DATA':
```
Comment (csabbee) [inline app/reducers/app-data.js]: nono!
There should be a tab!

File: app/reducers/edit-entry.js [hunk 3]
```diff
@@ -5,19 +5,19 @@ const defaultState = {
...
-            };
-        case 'SAVE_ENTRY': {
-            return {
-                editing: false,
-                underEdit: ''
-            };
-        }
-        default:
-            return state;
+    case 'UPDATE_USER':
```
Comment (csabbee) [inline app/reducers/edit-entry.js]: same here!

**Review by csabbee — CHANGES_REQUESTED** [review]

File: app/reducers/edit-entry.js (hunk 3 above)
Comment (ppseprus) [reply app/reducers/edit-entry.js]: Corrected. ESLint thrown an error and I set loose the auto-fixing option. Now the .eslintrc has a fix for the switch statements.

File: app/reducers/app-data.js (hunk 2 above)
Comment (ppseprus) [reply app/reducers/app-data.js]: Done...

File: app/actions/app-data.js (hunk 1 above)
Comment (ppseprus) [reply app/actions/app-data.js]: Exactly. The _user_ should not have the option to list all _users_. Why would he? I'd not change this for the time being, but if we need it in the future, I will put it back.

File: app/reducers/app-data.js [hunk 4]
```diff
@@ -1,30 +1,24 @@
...
+    userData: {}
 };
 
 export default function appData(state = defaultState, action) {
     switch (action.type) {
         case 'FETCHING_DATA':
             return {
                 isFetching: true,
-                users: state.users
+                userData: state.userData[0]
```
Comment (csabbee) [inline app/reducers/app-data.js]: you changed the userData to an object in the state. I would assume this will throw an error.

**Review by csabbee — CHANGES_REQUESTED** [review]

File: app/reducers/app-data.js (hunk 4 above)
Comment (ppseprus) [reply app/reducers/app-data.js]: You're right. I did not realise that the issue was in fact with how I validated the returned object. Empty object did not fail on `!userData`, therefore, it caused other issues. Now, the object validation is okas, so I can remove the `[0]`. Thanks for noticing!

**Review by csabbee — APPROVED** [review]

**Pull request merged**